In [1]:
import os
import pandas as pd
import numpy as np
import random
import warnings
random.seed(1) 
warnings.filterwarnings('ignore')
print('done importing')

done importing


### **Import and extract/read BRFSS2021 (CDC 2021 Health Indicators Survey) SAS (.XPT) file**

In [2]:
# Read SAS file
df= pd.read_sas("../Resources/LLCP2021.XPT")
df

_STATE  FMONTH        IDATE IMONTH   IDAY    IYEAR  DISPCODE  \
0          1.0     1.0  b'01192021'  b'01'  b'19'  b'2021'    1100.0   
1          1.0     1.0  b'01212021'  b'01'  b'21'  b'2021'    1100.0   
2          1.0     1.0  b'01212021'  b'01'  b'21'  b'2021'    1100.0   
3          1.0     1.0  b'01172021'  b'01'  b'17'  b'2021'    1100.0   
4          1.0     1.0  b'01152021'  b'01'  b'15'  b'2021'    1100.0   
...        ...     ...          ...    ...    ...      ...       ...   
438688    78.0    12.0  b'01062022'  b'01'  b'06'  b'2022'    1100.0   
438689    78.0    12.0  b'01122022'  b'01'  b'12'  b'2022'    1100.0   
438690    78.0    12.0  b'12212021'  b'12'  b'21'  b'2021'    1100.0   
438691    78.0    12.0  b'01112022'  b'01'  b'11'  b'2022'    1100.0   
438692    78.0    12.0  b'12222021'  b'12'  b'22'  b'2021'    1100.0   

                SEQNO          _PSU  CTELENM1  ...  _FRTRES1  _VEGRES1  \
0       b'2021000001'  2.021000e+09       1.0  ...       1.0       1.0   
1       b'2021000002'  2.021000e+09       1.0  ...       1.0       1.0   
2       b'2021000003'  2.021000e+09       1.0  ...       1.0       1.0   
3       b'2021000004'  2.021000e+09       1.0  ...       1.0       1.0   
4       b'2021000005'  2.021000e+09       1.0  ...       1.0       1.0   
...               ...           ...       ...  ...       ...       ...   
438688  b'2021001381'  2.021001e+09       NaN  ...       1.0       1.0   
438689  b'2021001382'  2.021001e+09       NaN  ...       1.0       1.0   
438690  b'2021001383'  2.021001e+09       NaN  ...       1.0       1.0   
438691  b'2021001384'  2.021001e+09       NaN  ...       1.0       1.0   
438692  b'2021001385'  2.021001e+09       NaN  ...       1.0       1.0   

        _FRUTSU1  _VEGESU1  _FRTLT1A  _VEGLT1A  _FRT16A  _VEG23A  \
0          100.0     214.0       1.0       1.0      1.0      1.0   
1          100.0     128.0       1.0       1.0      1.0      1.0   
2          100.0      71.0       1.0       2.0      1.0      1.0   
3          114.0     165.0       1.0       1.0      1.0      1.0   
4          100.0     258.0       1.0       1.0      1.0      1.0   
...          ...       ...       ...       ...      ...      ...   
438688     157.0     393.0       1.0       1.0      1.0      1.0   
438689     200.0     157.0       1.0       1.0      1.0      1.0   
438690     200.0     143.0       1.0       1.0      1.0      1.0   
438691     100.0     156.0       1.0       1.0      1.0      1.0   
438692      34.0      66.0       2.0       2.0      1.0      1.0   

            _FRUITE1      _VEGETE1  
0       5.397605e-79  5.397605e-79  
1       5.397605e-79  5.397605e-79  
2       5.397605e-79  5.397605e-79  
3       5.397605e-79  5.397605e-79  
4       5.397605e-79  5.397605e-79  
...              ...           ...  
438688  5.397605e-79  5.397605e-79  
438689  5.397605e-79  5.397605e-79  
438690  5.397605e-79  5.397605e-79  
438691  5.397605e-79  5.397605e-79  
438692  5.397605e-79  5.397605e-79  

[438693 rows x 303 columns]

In [3]:
# Eliminate fragmentation by creating a copy of DataFrame
df_bfrss2021 = df.copy()
df_bfrss2021

_STATE  FMONTH        IDATE IMONTH   IDAY    IYEAR  DISPCODE  \
0          1.0     1.0  b'01192021'  b'01'  b'19'  b'2021'    1100.0   
1          1.0     1.0  b'01212021'  b'01'  b'21'  b'2021'    1100.0   
2          1.0     1.0  b'01212021'  b'01'  b'21'  b'2021'    1100.0   
3          1.0     1.0  b'01172021'  b'01'  b'17'  b'2021'    1100.0   
4          1.0     1.0  b'01152021'  b'01'  b'15'  b'2021'    1100.0   
...        ...     ...          ...    ...    ...      ...       ...   
438688    78.0    12.0  b'01062022'  b'01'  b'06'  b'2022'    1100.0   
438689    78.0    12.0  b'01122022'  b'01'  b'12'  b'2022'    1100.0   
438690    78.0    12.0  b'12212021'  b'12'  b'21'  b'2021'    1100.0   
438691    78.0    12.0  b'01112022'  b'01'  b'11'  b'2022'    1100.0   
438692    78.0    12.0  b'12222021'  b'12'  b'22'  b'2021'    1100.0   

                SEQNO          _PSU  CTELENM1  ...  _FRTRES1  _VEGRES1  \
0       b'2021000001'  2.021000e+09       1.0  ...       1.0       1.0   
1       b'2021000002'  2.021000e+09       1.0  ...       1.0       1.0   
2       b'2021000003'  2.021000e+09       1.0  ...       1.0       1.0   
3       b'2021000004'  2.021000e+09       1.0  ...       1.0       1.0   
4       b'2021000005'  2.021000e+09       1.0  ...       1.0       1.0   
...               ...           ...       ...  ...       ...       ...   
438688  b'2021001381'  2.021001e+09       NaN  ...       1.0       1.0   
438689  b'2021001382'  2.021001e+09       NaN  ...       1.0       1.0   
438690  b'2021001383'  2.021001e+09       NaN  ...       1.0       1.0   
438691  b'2021001384'  2.021001e+09       NaN  ...       1.0       1.0   
438692  b'2021001385'  2.021001e+09       NaN  ...       1.0       1.0   

        _FRUTSU1  _VEGESU1  _FRTLT1A  _VEGLT1A  _FRT16A  _VEG23A  \
0          100.0     214.0       1.0       1.0      1.0      1.0   
1          100.0     128.0       1.0       1.0      1.0      1.0   
2          100.0      71.0       1.0       2.0      1.0      1.0   
3          114.0     165.0       1.0       1.0      1.0      1.0   
4          100.0     258.0       1.0       1.0      1.0      1.0   
...          ...       ...       ...       ...      ...      ...   
438688     157.0     393.0       1.0       1.0      1.0      1.0   
438689     200.0     157.0       1.0       1.0      1.0      1.0   
438690     200.0     143.0       1.0       1.0      1.0      1.0   
438691     100.0     156.0       1.0       1.0      1.0      1.0   
438692      34.0      66.0       2.0       2.0      1.0      1.0   

            _FRUITE1      _VEGETE1  
0       5.397605e-79  5.397605e-79  
1       5.397605e-79  5.397605e-79  
2       5.397605e-79  5.397605e-79  
3       5.397605e-79  5.397605e-79  
4       5.397605e-79  5.397605e-79  
...              ...           ...  
438688  5.397605e-79  5.397605e-79  
438689  5.397605e-79  5.397605e-79  
438690  5.397605e-79  5.397605e-79  
438691  5.397605e-79  5.397605e-79  
438692  5.397605e-79  5.397605e-79  

[438693 rows x 303 columns]

### **Define nature of the data**

In [4]:
# Get a list of all of the columns
df_bfrss2021.columns.values.tolist()

['_STATE',
 'FMONTH',
 'IDATE',
 'IMONTH',
 'IDAY',
 'IYEAR',
 'DISPCODE',
 'SEQNO',
 '_PSU',
 'CTELENM1',
 'PVTRESD1',
 'COLGHOUS',
 'STATERE1',
 'CELPHON1',
 'LADULT1',
 'COLGSEX',
 'NUMADULT',
 'LANDSEX',
 'NUMMEN',
 'NUMWOMEN',
 'RESPSLCT',
 'SAFETIME',
 'CTELNUM1',
 'CELLFON5',
 'CADULT1',
 'CELLSEX',
 'PVTRESD3',
 'CCLGHOUS',
 'CSTATE1',
 'LANDLINE',
 'HHADULT',
 'SEXVAR',
 'GENHLTH',
 'PHYSHLTH',
 'MENTHLTH',
 'POORHLTH',
 'PRIMINSR',
 'PERSDOC3',
 'MEDCOST1',
 'CHECKUP1',
 'EXERANY2',
 'BPHIGH6',
 'BPMEDS',
 'CHOLCHK3',
 'TOLDHI3',
 'CHOLMED3',
 'CVDINFR4',
 'CVDCRHD4',
 'CVDSTRK3',
 'ASTHMA3',
 'ASTHNOW',
 'CHCSCNCR',
 'CHCOCNCR',
 'CHCCOPD3',
 'ADDEPEV3',
 'CHCKDNY2',
 'DIABETE4',
 'DIABAGE3',
 'HAVARTH5',
 'ARTHEXER',
 'ARTHEDU',
 'LMTJOIN3',
 'ARTHDIS2',
 'JOINPAI2',
 'MARITAL',
 'EDUCA',
 'RENTHOM1',
 'NUMHHOL3',
 'NUMPHON3',
 'CPDEMO1B',
 'VETERAN3',
 'EMPLOY1',
 'CHILDREN',
 'INCOME3',
 'PREGNANT',
 'WEIGHT2',
 'HEIGHT3',
 'DEAF',
 'BLIND',
 'DECIDE',
 'DIFFWALK',
 'DIFF

In [5]:
# Determine the number of null values in each column
nan_sum = df_bfrss2021.isnull().sum()
pd.options.display.max_rows = 4000

print(nan_sum)

_STATE           0
FMONTH           0
IDATE            0
IMONTH           0
IDAY             0
IYEAR            0
DISPCODE         0
SEQNO            0
_PSU             0
CTELENM1    320907
PVTRESD1    320907
COLGHOUS    438663
STATERE1    320907
CELPHON1    320907
LADULT1     320907
COLGSEX     438663
NUMADULT    320926
LANDSEX     388847
NUMMEN      373255
NUMWOMEN    373255
RESPSLCT    374029
SAFETIME    117786
CTELNUM1    117787
CELLFON5    117786
CADULT1     117786
CELLSEX     117788
PVTRESD3    117786
CCLGHOUS    437438
CSTATE1     117786
LANDLINE    117786
HHADULT     117788
SEXVAR           0
GENHLTH          4
PHYSHLTH         3
MENTHLTH         2
POORHLTH    205279
PRIMINSR         3
PERSDOC3         2
MEDCOST1         5
CHECKUP1         2
EXERANY2         2
BPHIGH6          2
BPMEDS      266560
CHOLCHK3         2
TOLDHI3      60836
CHOLMED3     61571
CVDINFR4         2
CVDCRHD4         2
CVDSTRK3         2
ASTHMA3          2
ASTHNOW     377191
CHCSCNCR         2
CHCOCNCR    

In [6]:
# Percentage of missing values
nan_mean = df_bfrss2021.isnull().mean()
pd.options.display.max_rows = 4000

print(nan_mean)

_STATE      0.000000
FMONTH      0.000000
IDATE       0.000000
IMONTH      0.000000
IDAY        0.000000
IYEAR       0.000000
DISPCODE    0.000000
SEQNO       0.000000
_PSU        0.000000
CTELENM1    0.731507
PVTRESD1    0.731507
COLGHOUS    0.999932
STATERE1    0.731507
CELPHON1    0.731507
LADULT1     0.731507
COLGSEX     0.999932
NUMADULT    0.731550
LANDSEX     0.886376
NUMMEN      0.850834
NUMWOMEN    0.850834
RESPSLCT    0.852599
SAFETIME    0.268493
CTELNUM1    0.268495
CELLFON5    0.268493
CADULT1     0.268493
CELLSEX     0.268498
PVTRESD3    0.268493
CCLGHOUS    0.997139
CSTATE1     0.268493
LANDLINE    0.268493
HHADULT     0.268498
SEXVAR      0.000000
GENHLTH     0.000009
PHYSHLTH    0.000007
MENTHLTH    0.000005
POORHLTH    0.467933
PRIMINSR    0.000007
PERSDOC3    0.000005
MEDCOST1    0.000011
CHECKUP1    0.000005
EXERANY2    0.000005
BPHIGH6     0.000005
BPMEDS      0.607623
CHOLCHK3    0.000005
TOLDHI3     0.138676
CHOLMED3    0.140351
CVDINFR4    0.000005
CVDCRHD4    0

The majority of rows and columns include missing data.  As the dataset reflects the results if a comprehensive survey, one would expect to find a large number of missing data, and that the nature of the missing data would be highly variable and random.  Deleting rows or columns would remove most indicators.  The best approach is to fill in missing data.  Missing values are of the MCAR type, meaning there is absolutely no relationship between the data missing and any other observed or missing value in the dataset. In other words, those missing data points are a random subset of the dataset.  The most effective approach is to use a random sample imputer (using median) to fill in the missing data.


In [7]:
# Fill in NaN values with Random Sampler Imputation by Median
df_bfrss2021.fillna(df_bfrss2021.median())

_STATE  FMONTH        IDATE IMONTH   IDAY    IYEAR  DISPCODE  \
0          1.0     1.0  b'01192021'  b'01'  b'19'  b'2021'    1100.0   
1          1.0     1.0  b'01212021'  b'01'  b'21'  b'2021'    1100.0   
2          1.0     1.0  b'01212021'  b'01'  b'21'  b'2021'    1100.0   
3          1.0     1.0  b'01172021'  b'01'  b'17'  b'2021'    1100.0   
4          1.0     1.0  b'01152021'  b'01'  b'15'  b'2021'    1100.0   
...        ...     ...          ...    ...    ...      ...       ...   
438688    78.0    12.0  b'01062022'  b'01'  b'06'  b'2022'    1100.0   
438689    78.0    12.0  b'01122022'  b'01'  b'12'  b'2022'    1100.0   
438690    78.0    12.0  b'12212021'  b'12'  b'21'  b'2021'    1100.0   
438691    78.0    12.0  b'01112022'  b'01'  b'11'  b'2022'    1100.0   
438692    78.0    12.0  b'12222021'  b'12'  b'22'  b'2021'    1100.0   

                SEQNO          _PSU  CTELENM1  ...  _FRTRES1  _VEGRES1  \
0       b'2021000001'  2.021000e+09       1.0  ...       1.0       1.0   
1       b'2021000002'  2.021000e+09       1.0  ...       1.0       1.0   
2       b'2021000003'  2.021000e+09       1.0  ...       1.0       1.0   
3       b'2021000004'  2.021000e+09       1.0  ...       1.0       1.0   
4       b'2021000005'  2.021000e+09       1.0  ...       1.0       1.0   
...               ...           ...       ...  ...       ...       ...   
438688  b'2021001381'  2.021001e+09       1.0  ...       1.0       1.0   
438689  b'2021001382'  2.021001e+09       1.0  ...       1.0       1.0   
438690  b'2021001383'  2.021001e+09       1.0  ...       1.0       1.0   
438691  b'2021001384'  2.021001e+09       1.0  ...       1.0       1.0   
438692  b'2021001385'  2.021001e+09       1.0  ...       1.0       1.0   

        _FRUTSU1  _VEGESU1  _FRTLT1A  _VEGLT1A  _FRT16A  _VEG23A  \
0          100.0     214.0       1.0       1.0      1.0      1.0   
1          100.0     128.0       1.0       1.0      1.0      1.0   
2          100.0      71.0       1.0       2.0      1.0      1.0   
3          114.0     165.0       1.0       1.0      1.0      1.0   
4          100.0     258.0       1.0       1.0      1.0      1.0   
...          ...       ...       ...       ...      ...      ...   
438688     157.0     393.0       1.0       1.0      1.0      1.0   
438689     200.0     157.0       1.0       1.0      1.0      1.0   
438690     200.0     143.0       1.0       1.0      1.0      1.0   
438691     100.0     156.0       1.0       1.0      1.0      1.0   
438692      34.0      66.0       2.0       2.0      1.0      1.0   

            _FRUITE1      _VEGETE1  
0       5.397605e-79  5.397605e-79  
1       5.397605e-79  5.397605e-79  
2       5.397605e-79  5.397605e-79  
3       5.397605e-79  5.397605e-79  
4       5.397605e-79  5.397605e-79  
...              ...           ...  
438688  5.397605e-79  5.397605e-79  
438689  5.397605e-79  5.397605e-79  
438690  5.397605e-79  5.397605e-79  
438691  5.397605e-79  5.397605e-79  
438692  5.397605e-79  5.397605e-79  

[438693 rows x 303 columns]

### Using established research, we will narrow the available 303 indicators to reflect features that are known indicators for diabetes.

In [8]:
# Pre-diabetes diagnoses are incorporated in the DIABETE4 feature.  As a valuable indicator for diabetes, we will isolate this feature in a new column
# Add Prediabetic column
df_bfrss2021['PREDIAB'] = df_bfrss2021['DIABETE4']

In [9]:
# select chosen columns
brfss_df_selected = df_bfrss2021[['DISPCODE',
                                'DIABETE4',
                                'PREDIAB', 
                                'MEDCOST1',
                                '_PHYS14D',
                                '_MENT14D',
                                'EXERANY2',
                                'CVDINFR4',
                                'CVDCRHD4',
                                'CVDSTRK3',
                                'CHCCOPD3',
                                'CHCKDNY2',
                                'DIFFWALK',
                                '_IMPRACE',
                                '_RFHLTH',
                                '_HLTHPLN',
                                '_TOTINDA',
                                '_RFHYPE6',
                                '_SEX',
                                '_AGE_G',
                                '_RFBMI5',
                                '_EDUCAG',
                                '_INCOMG1',
                                '_SMOKER3',
                                '_RFBING5',
                                '_RFDRHV7',
                                '_FRTLT1A',
                                '_VEGLT1A',]]
brfss_df_selected

DISPCODE  DIABETE4  PREDIAB  MEDCOST1  _PHYS14D  _MENT14D  EXERANY2  \
0         1100.0       3.0      3.0       2.0       3.0       2.0       2.0   
1         1100.0       1.0      1.0       2.0       1.0       1.0       1.0   
2         1100.0       1.0      1.0       2.0       1.0       1.0       2.0   
3         1100.0       1.0      1.0       2.0       1.0       2.0       1.0   
4         1100.0       1.0      1.0       2.0       3.0       1.0       1.0   
...          ...       ...      ...       ...       ...       ...       ...   
438688    1100.0       2.0      2.0       1.0       1.0       3.0       1.0   
438689    1100.0       1.0      1.0       2.0       1.0       1.0       2.0   
438690    1100.0       3.0      3.0       2.0       1.0       1.0       1.0   
438691    1100.0       3.0      3.0       2.0       1.0       1.0       1.0   
438692    1100.0       3.0      3.0       2.0       2.0       2.0       1.0   

        CVDINFR4  CVDCRHD4  CVDSTRK3  ...  _SEX  _AGE_G  _RFBMI5  _EDUCAG  \
0            2.0       2.0       2.0  ...   2.0     6.0      1.0      2.0   
1            2.0       1.0       2.0  ...   2.0     6.0      9.0      4.0   
2            2.0       1.0       2.0  ...   2.0     6.0      2.0      2.0   
3            2.0       2.0       2.0  ...   2.0     5.0      2.0      2.0   
4            1.0       7.0       1.0  ...   1.0     6.0      2.0      1.0   
...          ...       ...       ...  ...   ...     ...      ...      ...   
438688       2.0       2.0       2.0  ...   2.0     2.0      1.0      2.0   
438689       2.0       2.0       2.0  ...   1.0     6.0      9.0      2.0   
438690       2.0       2.0       2.0  ...   1.0     4.0      2.0      4.0   
438691       2.0       2.0       2.0  ...   1.0     6.0      1.0      2.0   
438692       2.0       2.0       2.0  ...   2.0     4.0      2.0      4.0   

        _INCOMG1  _SMOKER3  _RFBING5  _RFDRHV7  _FRTLT1A  _VEGLT1A  
0            3.0       3.0       1.0       1.0       1.0       1.0  
1            9.0       4.0       1.0       1.0       1.0       1.0  
2            2.0       4.0       1.0       1.0       1.0       2.0  
3            5.0       4.0       2.0       1.0       1.0       1.0  
4            2.0       4.0       1.0       1.0       1.0       1.0  
...          ...       ...       ...       ...       ...       ...  
438688       3.0       3.0       1.0       1.0       1.0       1.0  
438689       1.0       4.0       1.0       1.0       1.0       1.0  
438690       6.0       4.0       1.0       1.0       1.0       1.0  
438691       4.0       4.0       1.0       1.0       1.0       1.0  
438692       4.0       4.0       1.0       1.0       2.0       2.0  

[438693 rows x 28 columns]

In [10]:
# The dataset includes an indicator for only partially collected survey results.
# As these are likely to skew our data, we will remove them in favour of creating models on a higher degree of complete data.
# Drop ny rows where survey has only been partially completed (dispcode = 1200)
brfss_df_selected.drop(brfss_df_selected[brfss_df_selected['DISPCODE'] >= 1200].index, inplace = True)
brfss_df_selected

DISPCODE  DIABETE4  PREDIAB  MEDCOST1  _PHYS14D  _MENT14D  EXERANY2  \
0         1100.0       3.0      3.0       2.0       3.0       2.0       2.0   
1         1100.0       1.0      1.0       2.0       1.0       1.0       1.0   
2         1100.0       1.0      1.0       2.0       1.0       1.0       2.0   
3         1100.0       1.0      1.0       2.0       1.0       2.0       1.0   
4         1100.0       1.0      1.0       2.0       3.0       1.0       1.0   
...          ...       ...      ...       ...       ...       ...       ...   
438688    1100.0       2.0      2.0       1.0       1.0       3.0       1.0   
438689    1100.0       1.0      1.0       2.0       1.0       1.0       2.0   
438690    1100.0       3.0      3.0       2.0       1.0       1.0       1.0   
438691    1100.0       3.0      3.0       2.0       1.0       1.0       1.0   
438692    1100.0       3.0      3.0       2.0       2.0       2.0       1.0   

        CVDINFR4  CVDCRHD4  CVDSTRK3  ...  _SEX  _AGE_G  _RFBMI5  _EDUCAG  \
0            2.0       2.0       2.0  ...   2.0     6.0      1.0      2.0   
1            2.0       1.0       2.0  ...   2.0     6.0      9.0      4.0   
2            2.0       1.0       2.0  ...   2.0     6.0      2.0      2.0   
3            2.0       2.0       2.0  ...   2.0     5.0      2.0      2.0   
4            1.0       7.0       1.0  ...   1.0     6.0      2.0      1.0   
...          ...       ...       ...  ...   ...     ...      ...      ...   
438688       2.0       2.0       2.0  ...   2.0     2.0      1.0      2.0   
438689       2.0       2.0       2.0  ...   1.0     6.0      9.0      2.0   
438690       2.0       2.0       2.0  ...   1.0     4.0      2.0      4.0   
438691       2.0       2.0       2.0  ...   1.0     6.0      1.0      2.0   
438692       2.0       2.0       2.0  ...   2.0     4.0      2.0      4.0   

        _INCOMG1  _SMOKER3  _RFBING5  _RFDRHV7  _FRTLT1A  _VEGLT1A  
0            3.0       3.0       1.0       1.0       1.0       1.0  
1            9.0       4.0       1.0       1.0       1.0       1.0  
2            2.0       4.0       1.0       1.0       1.0       2.0  
3            5.0       4.0       2.0       1.0       1.0       1.0  
4            2.0       4.0       1.0       1.0       1.0       1.0  
...          ...       ...       ...       ...       ...       ...  
438688       3.0       3.0       1.0       1.0       1.0       1.0  
438689       1.0       4.0       1.0       1.0       1.0       1.0  
438690       6.0       4.0       1.0       1.0       1.0       1.0  
438691       4.0       4.0       1.0       1.0       1.0       1.0  
438692       4.0       4.0       1.0       1.0       2.0       2.0  

[358899 rows x 28 columns]

In [11]:
# Drop DISPCODE column as it is no longer needed
brfss_df_selected = brfss_df_selected.drop('DISPCODE', axis=1)

In [12]:
# Create a copy of DataFrame
brfss_df_clean = brfss_df_selected.copy()
brfss_df_clean

DIABETE4  PREDIAB  MEDCOST1  _PHYS14D  _MENT14D  EXERANY2  CVDINFR4  \
0            3.0      3.0       2.0       3.0       2.0       2.0       2.0   
1            1.0      1.0       2.0       1.0       1.0       1.0       2.0   
2            1.0      1.0       2.0       1.0       1.0       2.0       2.0   
3            1.0      1.0       2.0       1.0       2.0       1.0       2.0   
4            1.0      1.0       2.0       3.0       1.0       1.0       1.0   
...          ...      ...       ...       ...       ...       ...       ...   
438688       2.0      2.0       1.0       1.0       3.0       1.0       2.0   
438689       1.0      1.0       2.0       1.0       1.0       2.0       2.0   
438690       3.0      3.0       2.0       1.0       1.0       1.0       2.0   
438691       3.0      3.0       2.0       1.0       1.0       1.0       2.0   
438692       3.0      3.0       2.0       2.0       2.0       1.0       2.0   

        CVDCRHD4  CVDSTRK3  CHCCOPD3  ...  _SEX  _AGE_G  _RFBMI5  _EDUCAG  \
0            2.0       2.0       1.0  ...   2.0     6.0      1.0      2.0   
1            1.0       2.0       2.0  ...   2.0     6.0      9.0      4.0   
2            1.0       2.0       2.0  ...   2.0     6.0      2.0      2.0   
3            2.0       2.0       2.0  ...   2.0     5.0      2.0      2.0   
4            7.0       1.0       2.0  ...   1.0     6.0      2.0      1.0   
...          ...       ...       ...  ...   ...     ...      ...      ...   
438688       2.0       2.0       1.0  ...   2.0     2.0      1.0      2.0   
438689       2.0       2.0       2.0  ...   1.0     6.0      9.0      2.0   
438690       2.0       2.0       2.0  ...   1.0     4.0      2.0      4.0   
438691       2.0       2.0       2.0  ...   1.0     6.0      1.0      2.0   
438692       2.0       2.0       2.0  ...   2.0     4.0      2.0      4.0   

        _INCOMG1  _SMOKER3  _RFBING5  _RFDRHV7  _FRTLT1A  _VEGLT1A  
0            3.0       3.0       1.0       1.0       1.0       1.0  
1            9.0       4.0       1.0       1.0       1.0       1.0  
2            2.0       4.0       1.0       1.0       1.0       2.0  
3            5.0       4.0       2.0       1.0       1.0       1.0  
4            2.0       4.0       1.0       1.0       1.0       1.0  
...          ...       ...       ...       ...       ...       ...  
438688       3.0       3.0       1.0       1.0       1.0       1.0  
438689       1.0       4.0       1.0       1.0       1.0       1.0  
438690       6.0       4.0       1.0       1.0       1.0       1.0  
438691       4.0       4.0       1.0       1.0       1.0       1.0  
438692       4.0       4.0       1.0       1.0       2.0       2.0  

[358899 rows x 27 columns]

In [13]:
# Print a list of the new columns
brfss_df_clean.columns.values.tolist()

['DIABETE4',
 'PREDIAB',
 'MEDCOST1',
 '_PHYS14D',
 '_MENT14D',
 'EXERANY2',
 'CVDINFR4',
 'CVDCRHD4',
 'CVDSTRK3',
 'CHCCOPD3',
 'CHCKDNY2',
 'DIFFWALK',
 '_IMPRACE',
 '_RFHLTH',
 '_HLTHPLN',
 '_TOTINDA',
 '_RFHYPE6',
 '_SEX',
 '_AGE_G',
 '_RFBMI5',
 '_EDUCAG',
 '_INCOMG1',
 '_SMOKER3',
 '_RFBING5',
 '_RFDRHV7',
 '_FRTLT1A',
 '_VEGLT1A']



```
# Cleaning our data

2021 BRFSS Codebook CDC: https://www.cdc.gov/brfss/annual_data/2021/pdf/codebook21_llcp-v2-508.pdf 
Codebook for the file showing variable name, location, and frequency of values for all reporting areas combined for the combined landline and cell phone data set.

In [14]:
# DIABETE4
# Make this ordinal: 0 = NO diabetes or only during pregnancy, 1 = YES for diabetes, 2 is for yes diabetes
# Remove all 7 (dont knows)
# Remove all 9 (refused)
brfss_df_clean['DIABETE4'] = brfss_df_clean['DIABETE4'].replace({2:0, 3:0, 4:0})
brfss_df_clean = brfss_df_clean[brfss_df_selected.DIABETE4 != 7]
brfss_df_clean = brfss_df_clean[brfss_df_selected.DIABETE4 != 9]
brfss_df_clean.groupby(['DIABETE4']).size()

DIABETE4
0.0    309494
1.0     48667
dtype: int64

In [15]:
# # PREDIAB
# # Create a duplicate colum for DIABETE4

brfss_df_clean['PREDIAB'] = brfss_df_clean['PREDIAB'].replace({1:0,2:0,3:0,4:1})
brfss_df_clean.groupby(['PREDIAB']).size()

PREDIAB
0.0    349661
1.0      8500
dtype: int64

In [16]:
# MEDCOST1
# Change 2 to 0 for no, 1 is already yes
# remove 7 for don't know and 9 for refused
brfss_df_clean['MEDCOST1'] = brfss_df_clean['MEDCOST1'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean.MEDCOST1 != 7]
brfss_df_clean = brfss_df_clean[brfss_df_clean.MEDCOST1 != 9]
brfss_df_clean.groupby(['MEDCOST1']).size()

MEDCOST1
0.0    331675
1.0     25570
dtype: int64

In [17]:
# _PHYS14D
# 1 = Zero days when physical health not good (Yes) and 0 = one or more days when physical health not good (No value 2 & 3)
# Remove all 9 (refused)
brfss_df_clean['_PHYS14D'] = brfss_df_clean['_PHYS14D'].replace({2:0, 3:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean._PHYS14D != 9]
brfss_df_clean.groupby(['_PHYS14D']).size()

_PHYS14D
0.0    116758
1.0    233313
dtype: int64

In [18]:
# _MENT14D
# 1 = Zero days when mental health not good (YES) and 0 = one or more days when mental health not good (NO value 2 & 3)
# Remove all 9 (refused)
brfss_df_clean['_MENT14D'] = brfss_df_clean['_MENT14D'].replace({2:0, 3:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean._MENT14D != 9]
brfss_df_clean.groupby(['_MENT14D']).size()

_MENT14D
0.0    128223
1.0    217364
dtype: int64

In [19]:
# EXERANY2
# Change 2 to 0 for no, 1 is already yes
# remove 7 for don't know and 9 for refused
brfss_df_clean['EXERANY2'] = brfss_df_clean['EXERANY2'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean.EXERANY2 != 7]
brfss_df_clean = brfss_df_clean[brfss_df_clean.EXERANY2 != 9]
brfss_df_clean.groupby(['EXERANY2']).size()

EXERANY2
0.0     82659
1.0    262449
dtype: int64

In [20]:
# CVDINFR4
# Change 2 to 0 for no, 1 is already yes
# remove 7 for don't know and 9 for refused
brfss_df_clean['CVDINFR4'] = brfss_df_clean['CVDINFR4'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean.CVDINFR4 != 7]
brfss_df_clean = brfss_df_clean[brfss_df_clean.CVDINFR4 != 9]
brfss_df_clean.groupby(['CVDINFR4']).size()

CVDINFR4
0.0    325497
1.0     17976
dtype: int64

In [21]:
# CVDCRHD4
# Change 2 to 0 for no, 1 is already yes
# remove 7 for don't know and 9 for refused
brfss_df_clean['CVDCRHD4'] = brfss_df_clean['CVDCRHD4'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean.CVDCRHD4 != 7]
brfss_df_clean = brfss_df_clean[brfss_df_clean.CVDCRHD4 != 9]
brfss_df_clean.groupby(['CVDCRHD4']).size()

CVDCRHD4
0.0    322636
1.0     18191
dtype: int64

In [22]:
# CVDSTRK3
# Change 2 to 0 for no, 1 is already yes
# remove 7 for don't know and 9 for refused
brfss_df_clean['CVDSTRK3'] = brfss_df_clean['CVDSTRK3'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean.CVDSTRK3 != 7]
brfss_df_clean = brfss_df_clean[brfss_df_clean.CVDSTRK3 != 9]
brfss_df_clean.groupby(['CVDSTRK3']).size()

CVDSTRK3
0.0    327285
1.0     12895
dtype: int64

In [23]:
# CHCCOPD3
# Change 2 to 0 for no, 1 is already yes
# remove 7 for don't know and 9 for refused
brfss_df_clean['CHCCOPD3'] = brfss_df_clean['CHCCOPD3'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean.CHCCOPD3 != 7]
brfss_df_clean = brfss_df_clean[brfss_df_clean.CHCCOPD3 != 9]
brfss_df_clean.groupby(['CHCCOPD3']).size()

CHCCOPD3
0.0    313211
1.0     25890
dtype: int64

In [24]:
# CHCKDNY2
# Change 2 to 0 for no, 1 is already yes
# remove 7 for don't know and 9 for refused
brfss_df_clean['CHCKDNY2'] = brfss_df_clean['CHCKDNY2'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean.CHCKDNY2 != 7]
brfss_df_clean = brfss_df_clean[brfss_df_clean.CHCKDNY2 != 9]
brfss_df_clean.groupby(['CHCKDNY2']).size()

CHCKDNY2
0.0    324954
1.0     13205
dtype: int64

In [25]:
# DIFFWALK
# Change 2 to 0 for no, 1 is already yes
# remove 7 for don't know and 9 for refused
brfss_df_clean['DIFFWALK'] = brfss_df_clean['DIFFWALK'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean.DIFFWALK != 7]
brfss_df_clean = brfss_df_clean[brfss_df_clean.DIFFWALK != 9]
brfss_df_clean.groupby(['DIFFWALK']).size()

DIFFWALK
0.0    285336
1.0     51812
dtype: int64

In [26]:
# _HLTHPLN
# Change 2 to 0 for no, 1 is already yes
# remove 9 for refused
brfss_df_clean['_HLTHPLN'] = brfss_df_clean['_HLTHPLN'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean._HLTHPLN != 9]
brfss_df_clean.groupby(['_HLTHPLN']).size()

_HLTHPLN
0.0     16481
1.0    309889
dtype: int64

In [27]:
# _TOTINDA
# Change 2 to 0 for no, 1 is already yes
# remove 9 for refused
brfss_df_clean['_TOTINDA'] = brfss_df_clean['_TOTINDA'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean._TOTINDA != 9]
brfss_df_clean.groupby(['_TOTINDA']).size()

_TOTINDA
0.0     76711
1.0    249659
dtype: int64

In [28]:
# _RFHYPE6
# Change 1 to 0 so it represetnts No high blood pressure and 2 to 1 so it represents high blood pressure
# remove 9 for refused
brfss_df_clean['_RFHYPE6'] = brfss_df_clean['_RFHYPE6'].replace({1:0, 2:1})
brfss_df_clean = brfss_df_clean[brfss_df_clean._RFHYPE6 != 9]
brfss_df_clean.groupby(['_RFHYPE6']).size()

_RFHYPE6
0.0    195964
1.0    129608
dtype: int64

In [29]:
# _RFBMI5
# Change 1 to 0 so it represetnts No high blood pressure and 2 to 1 so it represents high blood pressure
# remove 9 for refused
brfss_df_clean['_RFBMI5'] = brfss_df_clean['_RFBMI5'].replace({1:0, 2:1})
brfss_df_clean = brfss_df_clean[brfss_df_clean._RFBMI5 != 9]
brfss_df_clean.groupby(['_RFBMI5']).size()

_RFBMI5
0.0     93069
1.0    211500
dtype: int64

In [30]:
# _EDUCAG
# remove 9 for refused
brfss_df_clean = brfss_df_clean[brfss_df_clean._EDUCAG != 9]
brfss_df_clean.groupby(['_EDUCAG']).size()

_EDUCAG
1.0     15049
2.0     73740
3.0     84554
4.0    130700
dtype: int64

In [31]:
# _INCOMG1
# remove 9 for refused
brfss_df_clean = brfss_df_clean[brfss_df_clean._INCOMG1 != 9]
brfss_df_clean.groupby(['_INCOMG1']).size()

_INCOMG1
1.0    14974
2.0    24961
3.0    31241
4.0    36017
5.0    82788
6.0    53140
7.0    14926
dtype: int64

In [32]:
# _SMOKER3
# remove 9 for refused
brfss_df_clean = brfss_df_clean[brfss_df_clean._SMOKER3 != 9]
brfss_df_clean.groupby(['_SMOKER3']).size()

_SMOKER3
1.0     24630
2.0      9174
3.0     73059
4.0    150044
dtype: int64

In [33]:
# _RFBING5
# Change 1 to 0 so it represetnts No high blood pressure and 2 to 1 so it represents high blood pressure
# remove 9 for refused
brfss_df_clean['_RFBING5'] = brfss_df_clean['_RFBING5'].replace({1:0, 2:1})
brfss_df_clean = brfss_df_clean[brfss_df_clean._RFBING5 != 9]
brfss_df_clean.groupby(['_RFBING5']).size()

_RFBING5
0.0    217314
1.0     36513
dtype: int64

In [34]:
# _RFDRHV7
# Change 1 to 0 so it represetnts No high blood pressure and 2 to 1 so it represents high blood pressure
# remove 9 for refused
brfss_df_clean['_RFDRHV7'] = brfss_df_clean['_RFDRHV7'].replace({1:0, 2:1})
brfss_df_clean = brfss_df_clean[brfss_df_clean._RFDRHV7 != 9]
brfss_df_clean.groupby(['_RFDRHV7']).size()

_RFDRHV7
0.0    236835
1.0     15906
dtype: int64

In [35]:
# _FRTLT1A
# remove 9 for refused
brfss_df_clean['_FRTLT1A'] = brfss_df_clean['_FRTLT1A'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean._FRTLT1A != 9]
brfss_df_clean.groupby(['_FRTLT1A']).size()

_FRTLT1A
0.0     95661
1.0    152223
dtype: int64

In [36]:
# _VEGLT1A
# remove 9 for refused
brfss_df_clean['_VEGLT1A'] = brfss_df_clean['_VEGLT1A'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean._VEGLT1A != 9]
brfss_df_clean.groupby(['_VEGLT1A']).size()

_VEGLT1A
0.0     42836
1.0    200055
dtype: int64

In [37]:
# _SEX
# remove 9 for refused
brfss_df_clean['_SEX'] = brfss_df_clean['_SEX'].replace({1:0, 2:1})
brfss_df_clean.groupby(['_SEX']).size()

_SEX
0.0    116510
1.0    126381
dtype: int64

In [38]:
# _RFHLTH
# remove 9 for refused
brfss_df_clean['_RFHLTH'] = brfss_df_clean['_RFHLTH'].replace({2:0})
brfss_df_clean = brfss_df_clean[brfss_df_clean._VEGLT1A != 9]
brfss_df_clean.groupby(['_RFHLTH']).size()

_RFHLTH
0.0     36494
1.0    206199
9.0       198
dtype: int64

In [39]:
brfss_df_clean

DIABETE4  PREDIAB  MEDCOST1  _PHYS14D  _MENT14D  EXERANY2  CVDINFR4  \
0            0.0      0.0       0.0       0.0       0.0       0.0       0.0   
2            1.0      0.0       0.0       1.0       1.0       0.0       0.0   
3            1.0      0.0       0.0       1.0       0.0       1.0       0.0   
5            0.0      0.0       0.0       1.0       1.0       0.0       0.0   
9            0.0      0.0       0.0       0.0       0.0       1.0       0.0   
...          ...      ...       ...       ...       ...       ...       ...   
438687       1.0      0.0       0.0       1.0       1.0       1.0       0.0   
438688       0.0      0.0       1.0       1.0       0.0       1.0       0.0   
438690       0.0      0.0       0.0       1.0       1.0       1.0       0.0   
438691       0.0      0.0       0.0       1.0       1.0       1.0       0.0   
438692       0.0      0.0       0.0       0.0       0.0       1.0       0.0   

        CVDCRHD4  CVDSTRK3  CHCCOPD3  ...  _SEX  _AGE_G  _RFBMI5  _EDUCAG  \
0            0.0       0.0       1.0  ...   1.0     6.0      0.0      2.0   
2            1.0       0.0       0.0  ...   1.0     6.0      1.0      2.0   
3            0.0       0.0       0.0  ...   1.0     5.0      1.0      2.0   
5            0.0       0.0       1.0  ...   0.0     6.0      0.0      3.0   
9            0.0       0.0       0.0  ...   1.0     6.0      1.0      2.0   
...          ...       ...       ...  ...   ...     ...      ...      ...   
438687       0.0       0.0       0.0  ...   0.0     6.0      0.0      1.0   
438688       0.0       0.0       1.0  ...   1.0     2.0      0.0      2.0   
438690       0.0       0.0       0.0  ...   0.0     4.0      1.0      4.0   
438691       0.0       0.0       0.0  ...   0.0     6.0      0.0      2.0   
438692       0.0       0.0       0.0  ...   1.0     4.0      1.0      4.0   

        _INCOMG1  _SMOKER3  _RFBING5  _RFDRHV7  _FRTLT1A  _VEGLT1A  
0            3.0       3.0       0.0       0.0       1.0       1.0  
2            2.0       4.0       0.0       0.0       1.0       0.0  
3            5.0       4.0       1.0       0.0       1.0       1.0  
5            4.0       3.0       0.0       0.0       0.0       0.0  
9            5.0       3.0       0.0       0.0       1.0       1.0  
...          ...       ...       ...       ...       ...       ...  
438687       2.0       4.0       1.0       0.0       1.0       1.0  
438688       3.0       3.0       0.0       0.0       1.0       1.0  
438690       6.0       4.0       0.0       0.0       1.0       1.0  
438691       4.0       4.0       0.0       0.0       1.0       1.0  
438692       4.0       4.0       0.0       0.0       0.0       0.0  

[242891 rows x 27 columns]

In [40]:
#Rename the columns to make them more readable
brfss_table = brfss_df_clean.rename(columns = {'DIABETE4':'Diabetes',
                                            'PREDIAB':'Pre-Diabetic',    
                                            'MEDCOST1':'Afford Costs',
                                            '_PHYS14D':'Physical Health',
                                            '_MENT14D':'Mental Health',
                                            'EXERANY2':'Exercise 30 Days',
                                            'CVDINFR4':'Heart Attack',
                                            'CVDCRHD4':'Heart Disease',
                                            'CVDSTRK3':'Stroke',
                                            'CHCCOPD3':'Lung Disease',
                                            'CHCKDNY2':'Kidney Disease',
                                            'DIFFWALK':'Difficulty Walking',
                                            '_IMPRACE':'Race/Ethnicity',
                                            '_RFHLTH':'Overall Health',
                                            '_HLTHPLN':"Health Plan",
                                            '_TOTINDA':'Physical Activity',
                                            '_RFHYPE6':'High BP',
                                            '_SEX':'Gender',
                                            '_AGE_G':'Age',
                                            '_RFBMI5':'Overweight/Obese',
                                            '_EDUCAG':'Education',
                                            '_INCOMG1':'Income',
                                            '_SMOKER3':'Smoker',
                                            '_RFBING5':'Binge Drinker',
                                            '_RFDRHV7':'Heavy Drinker',
                                            '_FRTLT1A':'Fruit>1/Day',
                                            '_VEGLT1A':'Veggies>1/Day'})

brfss_table

Diabetes  Pre-Diabetic  Afford Costs  Physical Health  Mental Health  \
0            0.0           0.0           0.0              0.0            0.0   
2            1.0           0.0           0.0              1.0            1.0   
3            1.0           0.0           0.0              1.0            0.0   
5            0.0           0.0           0.0              1.0            1.0   
9            0.0           0.0           0.0              0.0            0.0   
...          ...           ...           ...              ...            ...   
438687       1.0           0.0           0.0              1.0            1.0   
438688       0.0           0.0           1.0              1.0            0.0   
438690       0.0           0.0           0.0              1.0            1.0   
438691       0.0           0.0           0.0              1.0            1.0   
438692       0.0           0.0           0.0              0.0            0.0   

        Exercise 30 Days  Heart Attack  Heart Disease  Stroke  Lung Disease  \
0                    0.0           0.0            0.0     0.0           1.0   
2                    0.0           0.0            1.0     0.0           0.0   
3                    1.0           0.0            0.0     0.0           0.0   
5                    0.0           0.0            0.0     0.0           1.0   
9                    1.0           0.0            0.0     0.0           0.0   
...                  ...           ...            ...     ...           ...   
438687               1.0           0.0            0.0     0.0           0.0   
438688               1.0           0.0            0.0     0.0           1.0   
438690               1.0           0.0            0.0     0.0           0.0   
438691               1.0           0.0            0.0     0.0           0.0   
438692               1.0           0.0            0.0     0.0           0.0   

        ...  Gender  Age  Overweight/Obese  Education  Income  Smoker  \
0       ...     1.0  6.0               0.0        2.0     3.0     3.0   
2       ...     1.0  6.0               1.0        2.0     2.0     4.0   
3       ...     1.0  5.0               1.0        2.0     5.0     4.0   
5       ...     0.0  6.0               0.0        3.0     4.0     3.0   
9       ...     1.0  6.0               1.0        2.0     5.0     3.0   
...     ...     ...  ...               ...        ...     ...     ...   
438687  ...     0.0  6.0               0.0        1.0     2.0     4.0   
438688  ...     1.0  2.0               0.0        2.0     3.0     3.0   
438690  ...     0.0  4.0               1.0        4.0     6.0     4.0   
438691  ...     0.0  6.0               0.0        2.0     4.0     4.0   
438692  ...     1.0  4.0               1.0        4.0     4.0     4.0   

        Binge Drinker  Heavy Drinker  Fruit>1/Day  Veggies>1/Day  
0                 0.0            0.0          1.0            1.0  
2                 0.0            0.0          1.0            0.0  
3                 1.0            0.0          1.0            1.0  
5                 0.0            0.0          0.0            0.0  
9                 0.0            0.0          1.0            1.0  
...               ...            ...          ...            ...  
438687            1.0            0.0          1.0            1.0  
438688            0.0            0.0          1.0            1.0  
438690            0.0            0.0          1.0            1.0  
438691            0.0            0.0          1.0            1.0  
438692            0.0            0.0          0.0            0.0  

[242891 rows x 27 columns]

In [41]:
brfss_table.dtypes

Diabetes              float64
Pre-Diabetic          float64
Afford Costs          float64
Physical Health       float64
Mental Health         float64
Exercise 30 Days      float64
Heart Attack          float64
Heart Disease         float64
Stroke                float64
Lung Disease          float64
Kidney Disease        float64
Difficulty Walking    float64
Race/Ethnicity        float64
Overall Health        float64
Health Plan           float64
Physical Activity     float64
High BP               float64
Gender                float64
Age                   float64
Overweight/Obese      float64
Education             float64
Income                float64
Smoker                float64
Binge Drinker         float64
Heavy Drinker         float64
Fruit>1/Day           float64
Veggies>1/Day         float64
dtype: object

In [42]:
# Change datatypes
brfss_table = brfss_table.fillna(0)
brfss_final = brfss_table.astype('int')
brfss_final

Diabetes  Pre-Diabetic  Afford Costs  Physical Health  Mental Health  \
0              0             0             0                0              0   
2              1             0             0                1              1   
3              1             0             0                1              0   
5              0             0             0                1              1   
9              0             0             0                0              0   
...          ...           ...           ...              ...            ...   
438687         1             0             0                1              1   
438688         0             0             1                1              0   
438690         0             0             0                1              1   
438691         0             0             0                1              1   
438692         0             0             0                0              0   

        Exercise 30 Days  Heart Attack  Heart Disease  Stroke  Lung Disease  \
0                      0             0              0       0             1   
2                      0             0              1       0             0   
3                      1             0              0       0             0   
5                      0             0              0       0             1   
9                      1             0              0       0             0   
...                  ...           ...            ...     ...           ...   
438687                 1             0              0       0             0   
438688                 1             0              0       0             1   
438690                 1             0              0       0             0   
438691                 1             0              0       0             0   
438692                 1             0              0       0             0   

        ...  Gender  Age  Overweight/Obese  Education  Income  Smoker  \
0       ...       1    6                 0          2       3       3   
2       ...       1    6                 1          2       2       4   
3       ...       1    5                 1          2       5       4   
5       ...       0    6                 0          3       4       3   
9       ...       1    6                 1          2       5       3   
...     ...     ...  ...               ...        ...     ...     ...   
438687  ...       0    6                 0          1       2       4   
438688  ...       1    2                 0          2       3       3   
438690  ...       0    4                 1          4       6       4   
438691  ...       0    6                 0          2       4       4   
438692  ...       1    4                 1          4       4       4   

        Binge Drinker  Heavy Drinker  Fruit>1/Day  Veggies>1/Day  
0                   0              0            1              1  
2                   0              0            1              0  
3                   1              0            1              1  
5                   0              0            0              0  
9                   0              0            1              1  
...               ...            ...          ...            ...  
438687              1              0            1              1  
438688              0              0            1              1  
438690              0              0            1              1  
438691              0              0            1              1  
438692              0              0            0              0  

[242891 rows x 27 columns]

In [44]:
# Save file to CSV 
brfss_final.to_csv('../Resources/diabetes_2021_health_indicators_BRFSS2021.csv', index=False)